pydantic_redis/model.py

```diff
@@ -198,20 +224,26 @@ def __get_serializable_dict(cls,
         Returns a dictionary that can be serialized.
         A few cleanups it does include:
           - Upserting any nested records in `record`
-          - Replacing the keys of nested records with their `__` suffixed versions e.g. `__author` instead of author
+          - Replacing the keys of nested records with their NESTED_MODEL_PREFIX suffixed versions e.g. `__author` instead of author
           - Replacing the values of nested records with their foreign keys
         """
         data = record.items() if isinstance(record, dict) else record
         new_data = {}
 
         for k, v in data:
             key, value = k, v
-
-            if isinstance(v, Model):
-                key = f"__{key}"
-                value = v.__class__.__insert_on_pipeline(
-                    _id=None, pipeline=pipeline, record=v, life_span=life_span)
-
+            if isinstance(value, List) and all(isinstance(item, Model) for item in value):
+                value = [item.__class__.__insert_on_pipeline(_id=None, pipeline=pipeline, record=item, life_span=life_span) for item in value]
+                key = f"{IN_LIST_NESTED_MODEL_PREFIX}{key}"
+            elif isinstance(value, Set) and all(isinstance(item, Model) for item in value):
+                value = {item.__class__.__insert_on_pipeline(_id=None, pipeline=pipeline, record=item, life_span=life_span) for item in value}
+                key = f"{IN_SET_NESTED_MODEL_PREFIX}{key}"
+            elif isinstance(value, Tuple) and all(isinstance(item, Model) for item in value):
+                value = *(item.__class__.__insert_on_pipeline(_id=None, pipeline=pipeline, record=item, life_span=life_span) for item in value),
```

Comment:
There is no tuple comprehension in python, so below code will create a *generator*
``` py
(item.__class__.__insert_on_pipeline(_id=None, pipeline=pipeline, record=item, life_span=life_span) for item in value)
```

Below code  will consume the generator (thanks to `*`) into a tuple (thanks to `,` at the end).
``` py 
*(item.__class__.__insert_on_pipeline(_id=None, pipeline=pipeline, record=item, life_span=life_span) for item in value),
```

It was interesting to find this out :)

Reply:
Hmmmn...interesting.

Reply:
Could we find out any trade-offs with regards to using generators in terms of speed of execution?

Reply:
Depends on the usage. Most of the time it is faster 

But as tuples handling may go in another pull request, we may have this conversation later

Reply:
Sure thing.